nucypher/policy/conditions/offchain.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Optional, Tuple
+
+import requests
+from jsonpath_ng.exceptions import JsonPathLexerError, JsonPathParserError
+from jsonpath_ng.ext import parse
+from marshmallow import ValidationError, fields, post_load, validate
+from marshmallow.fields import Field
+
+from nucypher.policy.conditions.base import AccessControlCondition
+from nucypher.policy.conditions.exceptions import (
+    ConditionEvaluationFailed,
+    InvalidCondition,
+)
+from nucypher.policy.conditions.lingo import ConditionType, ReturnValueTest
+from nucypher.policy.conditions.utils import CamelCaseSchema
+from nucypher.utilities.logging import Logger
+
+
+class JSONPathField(Field):
+    default_error_messages = {"invalid": "Not a valid JSONPath expression."}
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if not isinstance(value, str):
+            self.fail("invalid")
+        try:
+            parse(value)
+        except (JsonPathLexerError, JsonPathParserError):
+            self.fail("invalid")
+        return value
+
+
+class OffchainCondition(AccessControlCondition):
+    """
+    An offchain condition is a condition that can be evaluated by reading from a JSON
+    endpoint. This may be a REST service but the only requirement is that
+    the response is JSON and can be parsed using jsonpath.
+    """
+
+    CONDITION_TYPE = ConditionType.OFFCHAIN.value
+    LOGGER = Logger("nucypher.policy.conditions.offchain")
+
+    class Schema(CamelCaseSchema):
+
+        name = fields.Str(required=False)
+        condition_type = fields.Str(
+            validate=validate.Equal(ConditionType.OFFCHAIN.value), required=True
+        )
+        headers = fields.Dict(required=False)
+        parameters = fields.Dict(required=False)
+        endpoint = fields.Str(required=True)
+        query = JSONPathField(required=True)
```

Comment:
> (1) empty queries

Single, primitive values are valid JSON and therefore don't require a query path. E.g.: `42` is a valid JSON string.

> My general impression is that if the response body is empty then it's not a JSON API. 

Agreed, as `''` is not valid JSON.

Reply:
> Single, primitive values are valid JSON and therefore don't require a query path. E.g.: 42 is a valid JSON string.

The crux of the blank query question was more along these lines i.e. whether the "query" field should be required or not. If it is a required field we'd need to allow a non-existent query (blank or otherwise) for an endpoint that simply returns a value eg. 42.

Seems like "query" should *not* be a required field.